functions/src/definitions/userHandlers.js

```diff
@@ -83,6 +73,41 @@ exports.userHandlerWhatsapp = async function (message) {
         }
         break
       }
+      if (
+        normalizeSpaces(message.text.body) //normalise spaces needed cos of potential &nbsp when copying message on desktop whatsapp
+          .toLowerCase()
+          .startsWith(
+            "welcome to checkmate! send in this message to get started, and credit your friend with your referral. code:"
+          )
+      ) {
+        if (isFirstTimeUser) {
+          const code = message.text.body.split(":")[1].trim()
+          const referralSourceQuerySnap = await db
+            .collection("users")
+            .where("referralId", "==", code)
+            .limit(1)
+            .get()
+
+          if (referralSourceQuerySnap.empty) {
+            functions.logger.warn(
+              `Referral code ${code}, sent by ${from} not found`
+            )
+          } else {
+            const referralSourceSnap = referralSourceQuerySnap.docs[0]
+            await referralSourceSnap.ref.update({
+              referralCount: FieldValue.increment(1),
+            })
+          }
+        } else {
+          await sendWhatsappTextMessage(
```

Comment:
Is this message needed? Since we don't send any message to acknowledge successful referrels, we may not need to send this, but can log it somewhere instead.

Reply:
I feel it should be there, cos although there's no special message for successful referrals, they still get the newcomer welcome message with the menu. It's also to discourage ppl who think they can spam the referral msg to get their friends credited multiple times, haha.